OloEngine/src/OloEngine/Localization/LocalizationManager.cpp

```
@@ -0,0 +1,1229 @@
+        const char* envOrder[] = { "LC_ALL", "LC_MESSAGES", "LANG" };
+        for (const char* var : envOrder)
+        {
+            const char* v = std::getenv(var);
+            if (!v || !*v)
+                continue;
+            std::string s = v;
+            if (s == "C" || s == "POSIX")
+                continue;
+            // Drop encoding suffix (".UTF-8") and modifier ("@euro").
+            if (auto dot = s.find('.'); dot != std::string::npos)
+                s = s.substr(0, dot);
```

Cppcheck (2.20.0)

[performance] Ineffective call of function 'substr' because a prefix of the string is assigned to itself. Use resize() or pop_back() instead.

(uselessCallsSubstr)